test/helper.rb

```diff
@@ -45,6 +49,35 @@ def hit(uris)
   end
 end
 
+module WaitForServerLogs
+  def assert_io(re, io: @server, debug: false, timeout: 10)
+    regex = Regexp === re ? re : Regexp.new(Regexp.escape re)
+
+    STDOUT.puts('',
+      "#{' ' * 20}*** #{full_name} assert_io #{re} ***") if debug
+    l = ''
+
+    ttf = Time.now.to_f + 10
+
+    while IO.select([io], nil, nil, timeout) do
```

Comment:
I feel like we could use more descriptive variable names in this method, I think you slipped into writing C here 😆

Reply:
Yeah, I started writing that while frustrated.

Question: I've got STDERR going to the IO.pipe output for integration tests, and many of the other tests map Events IO to one IO.

Should the method dump all the read data to the console if the assert fails?